silo/src/new.re

```diff
@@ -0,0 +1,68 @@
+type error =
+  | InvalidPath(string)
+  | MustProvideName(string);
+exception NewError(error);
+
+let get_cwd = () => Fp.absoluteCurrentPlatformExn(Sys.getcwd());
+let derelativize = (~base=?, fname: Fp.firstClass) => {
+  switch (fname) {
+  | Absolute(path) => path
+  | Relative(path) =>
+    let b =
+      switch (base) {
+      | None => get_cwd()
+      | Some(path) => path
+      };
+    Fp.join(b, path);
+  };
+};
+
+let new_ = (~name=?, path) => {
+  let abs_path =
+    switch (Fp.testForPath(path)) {
+    | Some(path) => derelativize(path)
+    | None => raise(NewError(InvalidPath(path)))
+    };
+  let name =
+    switch (name) {
+    | Some(name) => name
+    | None =>
+      switch (Fp.baseName(abs_path)) {
+      | Some(name) => name
+      | None => raise(NewError(MustProvideName(path)))
+      }
+    };
+
```

Comment:
Should we check if the new project your trying to make already exists and provide a warning? (Maybe we can allow it to be overridden with a force argument if needed).